Address all the disabled warnings given by shell check

<!-- Please only use this template for submitting clean up requests -->

**What would you like to be cleaned**:
In https://github.com/kubernetes-sigs/kueue/pull/1552, We added shellcheck to the repository. Currently, we are ignoring alot of shellcheck warnings.

```shell
# Currently disabled these errors will take care of them later
DISABLED_ERRORS="SC2002,SC3028,SC3054,SC3014,SC3040,SC3046,SC3030,SC3010,SC3037,SC3045,SC3006,SC3018,SC3016,SC3011,SC3044,SC3043,SC3060,SC3024"
```

This issue tracks the progress we make on addressing all the disabled errors/warnings given by shellcheck. See https://github.com/kubernetes-sigs/kueue/pull/1552 for more information.
 
**Why is this needed**:
The goals of ShellCheck addressing all the warnings are:

- To point out and clarify typical beginner's syntax issues that cause a shell to give cryptic error messages.

- To point out and clarify typical intermediate level semantic problems that cause a shell to behave strangely and counter-intuitively.

- To point out subtle caveats, corner cases and pitfalls that may cause an advanced user's otherwise working script to fail under future circumstances.

Hi @tenzen-y 
Most of the disabled errors are warnings are already solved, see https://github.com/kubernetes-sigs/kueue/pull/1840#discussion_r1542880280

Currently, we are ignoring these errors/warnings, see [shellcheckrc](https://github.com/kubernetes-sigs/kueue/blob/main/.shellcheckrc)

```shell
# Currently disabled these errors will take care of them later

# Not following: (error message here)
# https://github.com/koalaman/shellcheck/wiki/SC1091
disable=SC1091
# Since you double quoted this, it will not word split, and the loop will only run once
# https://github.com/koalaman/shellcheck/wiki/SC2066
disable=SC2066
# Double quote to prevent globbing and word splitting
# https://github.com/koalaman/shellcheck/wiki/SC2086
disable=SC2086
# foo appears unused. Verify it or export it
# https://github.com/koalaman/shellcheck/wiki/SC2034
disable=SC2034
# This {/} is literal. Check if ; is missing or quote the expression.
# https://github.com/koalaman/shellcheck/wiki/SC1083
disable=SC1083
# Declare and assign separately to avoid masking return values
# https://github.com/koalaman/shellcheck/wiki/SC2155
disable=SC2155
# Quote this to prevent word splitting
# https://github.com/koalaman/shellcheck/wiki/SC2046
disable=SC2046
```

I checked and disabled all of these and ran `make shell-lint` and there were no errors 🎆 
should I go ahead and remove the [configuration where we are disabling them](https://github.com/kubernetes-sigs/kueue/blob/main/.shellcheckrc#L9-L31) in shell check?

> Hi @tenzen-y Most of the disabled errors and warnings are already solved and taken care of, see [#1840 (comment)](https://github.com/kubernetes-sigs/kueue/pull/1840#discussion_r1542880280)
> 
> Currently, we are ignoring these errors/warnings, see [shellcheckrc](https://github.com/kubernetes-sigs/kueue/blob/main/.shellcheckrc)
> 
> ```shell
> # Currently disabled these errors will take care of them later
> 
> # Not following: (error message here)
> # https://github.com/koalaman/shellcheck/wiki/SC1091
> disable=SC1091
> # Since you double quoted this, it will not word split, and the loop will only run once
> # https://github.com/koalaman/shellcheck/wiki/SC2066
> disable=SC2066
> # Double quote to prevent globbing and word splitting
> # https://github.com/koalaman/shellcheck/wiki/SC2086
> disable=SC2086
> # foo appears unused. Verify it or export it
> # https://github.com/koalaman/shellcheck/wiki/SC2034
> disable=SC2034
> # This {/} is literal. Check if ; is missing or quote the expression.
> # https://github.com/koalaman/shellcheck/wiki/SC1083
> disable=SC1083
> # Declare and assign separately to avoid masking return values
> # https://github.com/koalaman/shellcheck/wiki/SC2155
> disable=SC2155
> # Quote this to prevent word splitting
> # https://github.com/koalaman/shellcheck/wiki/SC2046
> disable=SC2046
> ```
> 
> I checked and disabled all of these and ran `make shell-lint` and there were no errors 🎆 should I go ahead and remove the [configuration where we are disabling them](https://github.com/kubernetes-sigs/kueue/blob/main/.shellcheckrc#L9-L31) in shell check?

Oh, thank you for investigating it!
Yeah, could you open PR to disable ignoring settings?

--- a/.shellcheckrc
+++ b/.shellcheckrc
@@ -9,8 +9,10 @@ external-sources=true
 # Currently disabled these errors will take care of them later
 
 # Not following: (error message here)
+# This variable is assigned to itself, so the assignment does nothing.
 # https://github.com/koalaman/shellcheck/wiki/SC2269
 disable=SC2269
+# Reasons include: file not found, no permissions, not included on the command line, not allowing shellcheck to follow files with -x, etc.
 # https://github.com/koalaman/shellcheck/wiki/SC1091
 disable=SC1091
 # Since you double quoted this, it will not word split, and the loop will only run once

--- a/.shellcheckrc
+++ b/.shellcheckrc
@@ -6,26 +6,3 @@
 # detect which files are failing
 external-sources=true
 
-# Currently disabled these errors will take care of them later
-
-# Not following: (error message here)
-# https://github.com/koalaman/shellcheck/wiki/SC1091
-disable=SC1091
-# Since you double quoted this, it will not word split, and the loop will only run once
-# https://github.com/koalaman/shellcheck/wiki/SC2066
-disable=SC2066
-# Double quote to prevent globbing and word splitting
-# https://github.com/koalaman/shellcheck/wiki/SC2086
-disable=SC2086
-# foo appears unused. Verify it or export it
-# https://github.com/koalaman/shellcheck/wiki/SC2034
-disable=SC2034
-# This {/} is literal. Check if ; is missing or quote the expression.
-# https://github.com/koalaman/shellcheck/wiki/SC1083
-disable=SC1083
-# Declare and assign separately to avoid masking return values
-# https://github.com/koalaman/shellcheck/wiki/SC2155
-disable=SC2155
-# Quote this to prevent word splitting
-# https://github.com/koalaman/shellcheck/wiki/SC2046
-disable=SC2046

--- a/.shellcheckrc
+++ b/.shellcheckrc
@@ -6,3 +6,28 @@
 # detect which files are failing
 external-sources=true
 
+# Currently disabled these errors will take care of them later
+
+# Not following: (error message here)
+# https://github.com/koalaman/shellcheck/wiki/SC2269
+disable=SC2269
+# https://github.com/koalaman/shellcheck/wiki/SC1091
+disable=SC1091
+# Since you double quoted this, it will not word split, and the loop will only run once
+# https://github.com/koalaman/shellcheck/wiki/SC2066
+disable=SC2066
+# Double quote to prevent globbing and word splitting
+# https://github.com/koalaman/shellcheck/wiki/SC2086
+disable=SC2086
+# foo appears unused. Verify it or export it
+# https://github.com/koalaman/shellcheck/wiki/SC2034
+disable=SC2034
+# This {/} is literal. Check if ; is missing or quote the expression.
+# https://github.com/koalaman/shellcheck/wiki/SC1083
+disable=SC1083
+# Declare and assign separately to avoid masking return values
+# https://github.com/koalaman/shellcheck/wiki/SC2155
+disable=SC2155
+# Quote this to prevent word splitting
+# https://github.com/koalaman/shellcheck/wiki/SC2046
+disable=SC2046

--- a/hack/verify-shellcheck.sh
+++ b/hack/verify-shellcheck.sh
@@ -17,21 +17,18 @@
 # allow overriding docker cli, which should work fine for this script
 DOCKER="${DOCKER:-docker}"
 
-
 SHELLCHECK_VERSION="0.9.0"
 SHELLCHECK_IMAGE="docker.io/koalaman/shellcheck-alpine:v0.9.0@sha256:e19ed93c22423970d56568e171b4512c9244fc75dd9114045016b4a0073ac4b7"
 
-scripts_to_check=("$@")
+# Initialize an empty array for scripts to check
+scripts_to_check=()
+
 if [[ "$#" == 0 ]]; then
-  # Find all shell scripts excluding:
-  # - Anything git-ignored - No need to lint untracked files.
-  # - ./_* - No need to lint output directories.
-  # - ./.git/* - Ignore anything in the git object store.
-  # - ./vendor* - Vendored code should be fixed upstream instead.
-  # - ./third_party/*, but re-include ./third_party/forked/*  - only code we
-  #    forked should be linted and fixed.
-  while IFS=$'\n' read -r script;
-    do git check-ignore -q "$script" || scripts_to_check+=("$script");
+  # Find all shell scripts excluding certain directories and patterns
+  while IFS=$'\n' read -r script; do
+    if ! git check-ignore -q "$script"; then
+      scripts_to_check+=("$script")
+    fi
   done < <(find . -name "*.sh" \
     -not \( \
       -path ./_\*      -o \
@@ -48,9 +45,13 @@ echo "Downloading ShellCheck Docker image..."
 # Run ShellCheck on all shell script files, excluding those in the 'vendor' directory.
 # Configuration loaded from the .shelcheckrc file.
 echo "Running ShellCheck..."
-"${DOCKER}" run \
-  --rm -v "$(pwd)" -w "$(pwd)" \
-    "${SHELLCHECK_IMAGE}" \
-  shellcheck "--color=${SHELLCHECK_COLORIZED_OUTPUT}" "${scripts_to_check[@]}" >&2 || res=$?
+if [ "${#scripts_to_check[@]}" -ne 0 ]; then
+  "${DOCKER}" run --rm -v "$(pwd)":/mnt -w /mnt "${SHELLCHECK_IMAGE}" shellcheck "${scripts_to_check[@]}" >&2 || res=$?
+else
+  echo "No scripts to check"
+  res=0
+fi
 
-echo "Shellcheck ran successfully"
+echo "Shellcheck ran successfully with exit code $res"
+
+exit $res

--- a/hack/verify-shellcheck.sh
+++ b/hack/verify-shellcheck.sh
@@ -46,12 +46,10 @@ echo "Downloading ShellCheck Docker image..."
 # Configuration loaded from the .shelcheckrc file.
 echo "Running ShellCheck..."
 if [ "${#scripts_to_check[@]}" -ne 0 ]; then
-  "${DOCKER}" run --rm -v "$(pwd)":/mnt -w /mnt "${SHELLCHECK_IMAGE}" shellcheck "${scripts_to_check[@]}" >&2 || res=$?
+  "${DOCKER}" run --rm -v "$(pwd)":/mnt -w /mnt "${SHELLCHECK_IMAGE}" shellcheck "${scripts_to_check[@]}" >&2 
 else
   echo "No scripts to check"
-  res=0
 fi
 
-echo "Shellcheck ran successfully with exit code $res"
+echo "Shellcheck ran successfully"
 
-exit $res

